### DISCUSSION TITLE
Regarding  What to Study

### DISCUSSION THREAD

**Discussion by Tanush103** [snigi-gupta/AttendingUniversityAtBuffalo#35] [General]
Hi,
Thanks for helping students always .I am recently joining UB this fall for MSCS and was having some doubts regarding the internships and jobs.
As you said to apply early for Internships ...I have applied to 2-3 companies who have opened internship opportunities for summer 22 (When I'll be eligible for Internships).But I have a doubt and can't decide what to do :-

1.In India while preparing for internships and jobs one must have cut throat DSA knowledge as well as CS Basics .I am currently more interested in Data Engineering  roles but I'm also inclining towards DSA and OOPS Java Since that is asked in coding interviews in FAANG and other product based companies.Like I have only 2 months left before I reach US , and being a Non CS Undergrad I have very little knowledge of Java and DSA OOPS and other core CS topics.So should I focus more on DSA and core CS Subjects for Interviews or should I go for learning Topics related to Data Engineering and Data Analytics??Can we get a job in US without medium -hard DSA Skills or like India it's a necessity while applying for Internships and jobs??

Thanks.

**Comment by snigi-gupta** [comment]
Hi @Tanush103 , 

Firstly, I apologize for the delay in replying to your questions. I think it is never too late, so here are my byte-sized inputs.

- **DSA is super important no matter what direction you choose in tech.** Read that again, and remember even if the job description doesn't ask for you to know DSA, knowing DSA will give you the edge in getting the job you are seeking.
- Before you arrive you should do the following in my opinion: Strengthen your python coding skills (heavily used at UB), you can also choose any other PL, but python is a good starting point for a non-cs student. Go through Strings, LinkedList, Arrays, Sets and Tuples, Dictionary etc. Also, go through basic Search & Sort algos. You can refer GeeksforGeeks, Abul Bari, back2swe, tushar roy videos on YT, also start a little bit of leetcode/hackerrank. start with easy questions only. Doing this before arriving will set a good foundation for you to keep up with the classes.
- For Data Engineering: you should know python or R (python more preferable in my opinion), the previous point would help you greatly. Specific to this field, look for some small projects on Kaggle. Also, check out some technical papers related to this field. I would highly recommend you to **_do what you love_** because that is what matters in the end. Even in a FAANG company, you can join as a Data Engineer, Data Analyst, and many other such roles.
- How do find what is your cup of tea? I would say explore. Go to your dream company's career page, look for the job descriptions that catch your eye/attention and then see what does this role demands. Start working on those.
- I didn't have a strong hands-on Java when I joined a FAANG company. I knew python, very well. And you learn on the go. So language does not matter, fundamentals do. Focus on those.

Feel free to post more questions, I would now answer all questions every Sunday. :)

**Reply by Tanush103** [reply]
Thanks a lot Snigdha..☺️
